### ISSUE TITLE
[valid-typeof] @typescript-eslint/recommended disables valid-typeof but it shouldn't

### ISSUE THREAD

**Issue by karoun** [typescript-eslint/typescript-eslint#3797]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "extends": ["plugin:@typescript-eslint/recommended"]
}
```

```TS
typeof window === undefined
typeof window === null
```

**Expected Result**

```
error  Typeof comparisons should be to string literals  valid-typeof
```

**Actual Result**

🦗 

**Additional Info**

The `@typescript-eslint/recommended` [ruleset disables the `valid-typeof` ESLint rule](https://github.com/typescript-eslint/typescript-eslint/blob/043462ef4d8f3aa16c10efb84fea25a193deefac/packages/eslint-plugin/src/configs/eslint-recommended.ts#L31) on [the basis that it's covered by TypeScript](https://github.com/microsoft/TypeScript/issues/27910)'s "This condition will always return 'false'" handling.

However, this is incorrect: [TypeScript does not detect this incorrect condition](https://github.com/microsoft/TypeScript/issues/29200) in the case of `null` and `undefined`, and so the `valid-typeof` rule should be re-enabled in the `recommended` preset.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.23.0` |
| `@typescript-eslint/parser`        | `4.23.0` |
| `TypeScript`                       | `4.2.4` |
| `ESLint`                           | `7.26.0` |
| `node`                             | `12.13.1` |

**Comment by JounQin** [comment]
A related proof https://github.com/typescript-eslint/typescript-eslint/pull/3635

**Comment by tanohzana** [comment]
I'm working on it!
